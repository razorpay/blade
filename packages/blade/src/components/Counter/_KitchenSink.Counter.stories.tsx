import { composeStories } from '@storybook/react';
import * as counterStories from './Counter.stories';
import { Box } from '~components/Box';
import { Heading } from '~components/Typography';

const allStories = Object.values(composeStories(counterStories));

export const Counter = (): JSX.Element => {
  return (
    <Box display="flex" flexDirection="column" gap="spacing.4">
      {allStories.map((Story) => {
        return (
          <>
            <Heading>{Story.storyName}</Heading>
            <Story />
          </>
        );
      })}
    </Box>
  );
};

export default {
  title: 'Components/KitchenSink/Counter',
  component: Counter,
  parameters: {
    // enable Chromatic's snapshotting only for kitchensink
    chromatic: { disableSnapshot: false },
    options: { showPanel: false },
  },
};
