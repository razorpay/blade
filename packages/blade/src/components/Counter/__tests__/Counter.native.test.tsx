import { Counter } from '../Counter';
import renderWithTheme from '~src/_helpers/testing/renderWithTheme.native';

describe('<Counter />', () => {
  it('should render Counter with default props', () => {
    const { toJSON } = renderWithTheme(<Counter value={20} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should limit value with max prop', () => {
    const { getByText } = renderWithTheme(<Counter value={20} max={10} />);
    expect(getByText('10+')).toBeTruthy();
  });

  it('should render small size Counter', () => {
    const { toJSON } = renderWithTheme(<Counter size="small" value={20} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render medium size Counter', () => {
    const { toJSON } = renderWithTheme(<Counter size="medium" value={20} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render low contrast positive intent Counter', () => {
    const { toJSON } = renderWithTheme(<Counter intent="positive" contrast="low" value={20} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render high contrast positive intent Counter', () => {
    const { toJSON } = renderWithTheme(<Counter intent="positive" contrast="high" value={20} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render low contrast negative intent Counter', () => {
    const { toJSON } = renderWithTheme(<Counter intent="negative" contrast="low" value={20} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render high contrast negative intent Counter', () => {
    const { toJSON } = renderWithTheme(<Counter intent="negative" contrast="high" value={20} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render low contrast notice intent Counter', () => {
    const { toJSON } = renderWithTheme(<Counter intent="notice" contrast="low" value={20} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render high contrast notice intent Counter', () => {
    const { toJSON } = renderWithTheme(<Counter intent="notice" contrast="high" value={20} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render low contrast information intent Counter', () => {
    const { toJSON } = renderWithTheme(<Counter intent="information" contrast="low" value={20} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render high contrast information intent Counter', () => {
    const { toJSON } = renderWithTheme(<Counter intent="information" contrast="high" value={20} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render low contrast neutral intent Counter', () => {
    const { toJSON } = renderWithTheme(<Counter intent="neutral" contrast="low" value={20} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render high contrast neutral intent Counter', () => {
    const { toJSON } = renderWithTheme(<Counter intent="neutral" contrast="high" value={20} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
