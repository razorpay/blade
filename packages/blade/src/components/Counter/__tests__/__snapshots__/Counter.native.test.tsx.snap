// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> should render Counter with default props 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    backgroundColor="feedback.background.neutral.lowContrast"
    data-blade-component="base-box"
    platform="onMobile"
    size="medium"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "hsla(216, 15%, 54%, 0.09)",
          "borderRadius": 9999,
          "display": "flex",
          "flexWrap": "nowrap",
          "maxWidth": 100,
        },
      ]
    }
  >
    <View
      alignItems="center"
      data-blade-component="counter"
      display="flex"
      flexDirection="row"
      justifyContent="center"
      overflow="hidden"
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingBottom": 2,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 2,
          },
        ]
      }
    >
      <Text
        color="feedback.text.neutral.lowContrast"
        data-blade-component="text"
        fontFamily="text"
        fontSize={75}
        fontStyle="normal"
        fontWeight="regular"
        lineHeight={50}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "hsla(216, 27%, 36%, 1)",
              "fontFamily": "Lato",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 16,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "textAlign": "center",
              "textDecorationLine": "none",
            },
          ]
        }
        textAlign="center"
      >
        20
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Counter /> should render high contrast blue variant Counter 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    backgroundColor="badge.background.blue.highContrast"
    data-blade-component="base-box"
    platform="onMobile"
    size="medium"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "hsla(218, 89%, 51%, 1)",
          "borderRadius": 9999,
          "display": "flex",
          "flexWrap": "nowrap",
          "maxWidth": 100,
        },
      ]
    }
  >
    <View
      alignItems="center"
      data-blade-component="counter"
      display="flex"
      flexDirection="row"
      justifyContent="center"
      overflow="hidden"
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingBottom": 2,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 2,
          },
        ]
      }
    >
      <Text
        color="badge.text.blue.highContrast"
        data-blade-component="text"
        fontFamily="text"
        fontSize={75}
        fontStyle="normal"
        fontWeight="regular"
        lineHeight={50}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "hsla(0, 0%, 100%, 1)",
              "fontFamily": "Lato",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 16,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "textAlign": "center",
              "textDecorationLine": "none",
            },
          ]
        }
        textAlign="center"
      >
        20
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Counter /> should render high contrast information variant Counter 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    backgroundColor="feedback.background.information.highContrast"
    data-blade-component="base-box"
    platform="onMobile"
    size="medium"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "hsla(193, 100%, 35%, 1)",
          "borderRadius": 9999,
          "display": "flex",
          "flexWrap": "nowrap",
          "maxWidth": 100,
        },
      ]
    }
  >
    <View
      alignItems="center"
      data-blade-component="counter"
      display="flex"
      flexDirection="row"
      justifyContent="center"
      overflow="hidden"
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingBottom": 2,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 2,
          },
        ]
      }
    >
      <Text
        color="feedback.text.information.highContrast"
        data-blade-component="text"
        fontFamily="text"
        fontSize={75}
        fontStyle="normal"
        fontWeight="regular"
        lineHeight={50}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "hsla(0, 0%, 100%, 1)",
              "fontFamily": "Lato",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 16,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "textAlign": "center",
              "textDecorationLine": "none",
            },
          ]
        }
        textAlign="center"
      >
        20
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Counter /> should render high contrast negative variant Counter 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    backgroundColor="feedback.background.negative.highContrast"
    data-blade-component="base-box"
    platform="onMobile"
    size="medium"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "hsla(8, 73%, 47%, 1)",
          "borderRadius": 9999,
          "display": "flex",
          "flexWrap": "nowrap",
          "maxWidth": 100,
        },
      ]
    }
  >
    <View
      alignItems="center"
      data-blade-component="counter"
      display="flex"
      flexDirection="row"
      justifyContent="center"
      overflow="hidden"
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingBottom": 2,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 2,
          },
        ]
      }
    >
      <Text
        color="feedback.text.negative.highContrast"
        data-blade-component="text"
        fontFamily="text"
        fontSize={75}
        fontStyle="normal"
        fontWeight="regular"
        lineHeight={50}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "hsla(0, 0%, 100%, 1)",
              "fontFamily": "Lato",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 16,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "textAlign": "center",
              "textDecorationLine": "none",
            },
          ]
        }
        textAlign="center"
      >
        20
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Counter /> should render high contrast neutral variant Counter 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    backgroundColor="feedback.background.neutral.highContrast"
    data-blade-component="base-box"
    platform="onMobile"
    size="medium"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "hsla(216, 33%, 29%, 1)",
          "borderRadius": 9999,
          "display": "flex",
          "flexWrap": "nowrap",
          "maxWidth": 100,
        },
      ]
    }
  >
    <View
      alignItems="center"
      data-blade-component="counter"
      display="flex"
      flexDirection="row"
      justifyContent="center"
      overflow="hidden"
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingBottom": 2,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 2,
          },
        ]
      }
    >
      <Text
        color="feedback.text.neutral.highContrast"
        data-blade-component="text"
        fontFamily="text"
        fontSize={75}
        fontStyle="normal"
        fontWeight="regular"
        lineHeight={50}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "hsla(0, 0%, 100%, 1)",
              "fontFamily": "Lato",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 16,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "textAlign": "center",
              "textDecorationLine": "none",
            },
          ]
        }
        textAlign="center"
      >
        20
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Counter /> should render high contrast notice variant Counter 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    backgroundColor="feedback.background.notice.highContrast"
    data-blade-component="base-box"
    platform="onMobile"
    size="medium"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "hsla(38, 97%, 38%, 1)",
          "borderRadius": 9999,
          "display": "flex",
          "flexWrap": "nowrap",
          "maxWidth": 100,
        },
      ]
    }
  >
    <View
      alignItems="center"
      data-blade-component="counter"
      display="flex"
      flexDirection="row"
      justifyContent="center"
      overflow="hidden"
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingBottom": 2,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 2,
          },
        ]
      }
    >
      <Text
        color="feedback.text.notice.highContrast"
        data-blade-component="text"
        fontFamily="text"
        fontSize={75}
        fontStyle="normal"
        fontWeight="regular"
        lineHeight={50}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "hsla(0, 0%, 100%, 1)",
              "fontFamily": "Lato",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 16,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "textAlign": "center",
              "textDecorationLine": "none",
            },
          ]
        }
        textAlign="center"
      >
        20
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Counter /> should render high contrast positive variant Counter 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    backgroundColor="feedback.background.positive.highContrast"
    data-blade-component="base-box"
    platform="onMobile"
    size="medium"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "hsla(160, 100%, 26%, 1)",
          "borderRadius": 9999,
          "display": "flex",
          "flexWrap": "nowrap",
          "maxWidth": 100,
        },
      ]
    }
  >
    <View
      alignItems="center"
      data-blade-component="counter"
      display="flex"
      flexDirection="row"
      justifyContent="center"
      overflow="hidden"
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingBottom": 2,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 2,
          },
        ]
      }
    >
      <Text
        color="feedback.text.positive.highContrast"
        data-blade-component="text"
        fontFamily="text"
        fontSize={75}
        fontStyle="normal"
        fontWeight="regular"
        lineHeight={50}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "hsla(0, 0%, 100%, 1)",
              "fontFamily": "Lato",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 16,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "textAlign": "center",
              "textDecorationLine": "none",
            },
          ]
        }
        textAlign="center"
      >
        20
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Counter /> should render large size Counter 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    backgroundColor="feedback.background.neutral.lowContrast"
    data-blade-component="base-box"
    platform="onMobile"
    size="large"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "hsla(216, 15%, 54%, 0.09)",
          "borderRadius": 9999,
          "display": "flex",
          "flexWrap": "nowrap",
          "maxWidth": 100,
        },
      ]
    }
  >
    <View
      alignItems="center"
      data-blade-component="counter"
      display="flex"
      flexDirection="row"
      justifyContent="center"
      overflow="hidden"
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingBottom": 2,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 2,
          },
        ]
      }
    >
      <Text
        color="feedback.text.neutral.lowContrast"
        data-blade-component="text"
        fontFamily="text"
        fontSize={100}
        fontStyle="normal"
        fontWeight="regular"
        lineHeight={100}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "hsla(216, 27%, 36%, 1)",
              "fontFamily": "Lato",
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 20,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "textAlign": "center",
              "textDecorationLine": "none",
            },
          ]
        }
        textAlign="center"
      >
        20
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Counter /> should render low contrast blue variant Counter 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    backgroundColor="badge.background.blue.lowContrast"
    data-blade-component="base-box"
    platform="onMobile"
    size="medium"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "hsla(218, 89%, 51%, 0.09)",
          "borderRadius": 9999,
          "display": "flex",
          "flexWrap": "nowrap",
          "maxWidth": 100,
        },
      ]
    }
  >
    <View
      alignItems="center"
      data-blade-component="counter"
      display="flex"
      flexDirection="row"
      justifyContent="center"
      overflow="hidden"
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingBottom": 2,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 2,
          },
        ]
      }
    >
      <Text
        color="badge.text.blue.lowContrast"
        data-blade-component="text"
        fontFamily="text"
        fontSize={75}
        fontStyle="normal"
        fontWeight="regular"
        lineHeight={50}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "hsla(218, 89%, 51%, 1)",
              "fontFamily": "Lato",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 16,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "textAlign": "center",
              "textDecorationLine": "none",
            },
          ]
        }
        textAlign="center"
      >
        20
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Counter /> should render low contrast information variant Counter 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    backgroundColor="feedback.background.information.lowContrast"
    data-blade-component="base-box"
    platform="onMobile"
    size="medium"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "hsla(193, 100%, 35%, 0.09)",
          "borderRadius": 9999,
          "display": "flex",
          "flexWrap": "nowrap",
          "maxWidth": 100,
        },
      ]
    }
  >
    <View
      alignItems="center"
      data-blade-component="counter"
      display="flex"
      flexDirection="row"
      justifyContent="center"
      overflow="hidden"
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingBottom": 2,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 2,
          },
        ]
      }
    >
      <Text
        color="feedback.text.information.lowContrast"
        data-blade-component="text"
        fontFamily="text"
        fontSize={75}
        fontStyle="normal"
        fontWeight="regular"
        lineHeight={50}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "hsla(193, 100%, 35%, 1)",
              "fontFamily": "Lato",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 16,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "textAlign": "center",
              "textDecorationLine": "none",
            },
          ]
        }
        textAlign="center"
      >
        20
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Counter /> should render low contrast negative variant Counter 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    backgroundColor="feedback.background.negative.lowContrast"
    data-blade-component="base-box"
    platform="onMobile"
    size="medium"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "hsla(9, 91%, 56%, 0.09)",
          "borderRadius": 9999,
          "display": "flex",
          "flexWrap": "nowrap",
          "maxWidth": 100,
        },
      ]
    }
  >
    <View
      alignItems="center"
      data-blade-component="counter"
      display="flex"
      flexDirection="row"
      justifyContent="center"
      overflow="hidden"
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingBottom": 2,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 2,
          },
        ]
      }
    >
      <Text
        color="feedback.text.negative.lowContrast"
        data-blade-component="text"
        fontFamily="text"
        fontSize={75}
        fontStyle="normal"
        fontWeight="regular"
        lineHeight={50}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "hsla(8, 73%, 47%, 1)",
              "fontFamily": "Lato",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 16,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "textAlign": "center",
              "textDecorationLine": "none",
            },
          ]
        }
        textAlign="center"
      >
        20
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Counter /> should render low contrast neutral variant Counter 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    backgroundColor="feedback.background.neutral.lowContrast"
    data-blade-component="base-box"
    platform="onMobile"
    size="medium"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "hsla(216, 15%, 54%, 0.09)",
          "borderRadius": 9999,
          "display": "flex",
          "flexWrap": "nowrap",
          "maxWidth": 100,
        },
      ]
    }
  >
    <View
      alignItems="center"
      data-blade-component="counter"
      display="flex"
      flexDirection="row"
      justifyContent="center"
      overflow="hidden"
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingBottom": 2,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 2,
          },
        ]
      }
    >
      <Text
        color="feedback.text.neutral.lowContrast"
        data-blade-component="text"
        fontFamily="text"
        fontSize={75}
        fontStyle="normal"
        fontWeight="regular"
        lineHeight={50}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "hsla(216, 27%, 36%, 1)",
              "fontFamily": "Lato",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 16,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "textAlign": "center",
              "textDecorationLine": "none",
            },
          ]
        }
        textAlign="center"
      >
        20
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Counter /> should render low contrast notice variant Counter 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    backgroundColor="feedback.background.notice.lowContrast"
    data-blade-component="base-box"
    platform="onMobile"
    size="medium"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "hsla(36, 100%, 44%, 0.09)",
          "borderRadius": 9999,
          "display": "flex",
          "flexWrap": "nowrap",
          "maxWidth": 100,
        },
      ]
    }
  >
    <View
      alignItems="center"
      data-blade-component="counter"
      display="flex"
      flexDirection="row"
      justifyContent="center"
      overflow="hidden"
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingBottom": 2,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 2,
          },
        ]
      }
    >
      <Text
        color="feedback.text.notice.lowContrast"
        data-blade-component="text"
        fontFamily="text"
        fontSize={75}
        fontStyle="normal"
        fontWeight="regular"
        lineHeight={50}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "hsla(38, 97%, 38%, 1)",
              "fontFamily": "Lato",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 16,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "textAlign": "center",
              "textDecorationLine": "none",
            },
          ]
        }
        textAlign="center"
      >
        20
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Counter /> should render low contrast positive variant Counter 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    backgroundColor="feedback.background.positive.lowContrast"
    data-blade-component="base-box"
    platform="onMobile"
    size="medium"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "hsla(155, 100%, 31%, 0.09)",
          "borderRadius": 9999,
          "display": "flex",
          "flexWrap": "nowrap",
          "maxWidth": 100,
        },
      ]
    }
  >
    <View
      alignItems="center"
      data-blade-component="counter"
      display="flex"
      flexDirection="row"
      justifyContent="center"
      overflow="hidden"
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingBottom": 2,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 2,
          },
        ]
      }
    >
      <Text
        color="feedback.text.positive.lowContrast"
        data-blade-component="text"
        fontFamily="text"
        fontSize={75}
        fontStyle="normal"
        fontWeight="regular"
        lineHeight={50}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "hsla(160, 100%, 26%, 1)",
              "fontFamily": "Lato",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 16,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "textAlign": "center",
              "textDecorationLine": "none",
            },
          ]
        }
        textAlign="center"
      >
        20
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Counter /> should render medium size Counter 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    backgroundColor="feedback.background.neutral.lowContrast"
    data-blade-component="base-box"
    platform="onMobile"
    size="medium"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "hsla(216, 15%, 54%, 0.09)",
          "borderRadius": 9999,
          "display": "flex",
          "flexWrap": "nowrap",
          "maxWidth": 100,
        },
      ]
    }
  >
    <View
      alignItems="center"
      data-blade-component="counter"
      display="flex"
      flexDirection="row"
      justifyContent="center"
      overflow="hidden"
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingBottom": 2,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 2,
          },
        ]
      }
    >
      <Text
        color="feedback.text.neutral.lowContrast"
        data-blade-component="text"
        fontFamily="text"
        fontSize={75}
        fontStyle="normal"
        fontWeight="regular"
        lineHeight={50}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "hsla(216, 27%, 36%, 1)",
              "fontFamily": "Lato",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 16,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "textAlign": "center",
              "textDecorationLine": "none",
            },
          ]
        }
        textAlign="center"
      >
        20
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Counter /> should render small size Counter 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    backgroundColor="feedback.background.neutral.lowContrast"
    data-blade-component="base-box"
    platform="onMobile"
    size="small"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "hsla(216, 15%, 54%, 0.09)",
          "borderRadius": 9999,
          "display": "flex",
          "flexWrap": "nowrap",
          "maxWidth": 100,
        },
      ]
    }
  >
    <View
      alignItems="center"
      data-blade-component="counter"
      display="flex"
      flexDirection="row"
      justifyContent="center"
      overflow="hidden"
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingBottom": 0,
            "paddingLeft": 4,
            "paddingRight": 4,
            "paddingTop": 0,
          },
        ]
      }
    >
      <Text
        color="feedback.text.neutral.lowContrast"
        data-blade-component="text"
        fontFamily="text"
        fontSize={25}
        fontStyle="normal"
        fontWeight="regular"
        lineHeight={50}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "hsla(216, 27%, 36%, 1)",
              "fontFamily": "Lato",
              "fontSize": 10,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 16,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "textAlign": "center",
              "textDecorationLine": "none",
            },
          ]
        }
        textAlign="center"
      >
        20
      </Text>
    </View>
  </View>
</View>
`;
